pkg/apis/extensions/types.go

```diff
@@ -899,3 +899,45 @@ type NetworkPolicyList struct {
 
 	Items []NetworkPolicy `json:"items"`
 }
+
+// +genclient=true
+// +nonNamespaced=true
+
+// StorageClass describes the parameters for a class of storage for
+// which PersistentVolumes can be dynamically provisioned.
+//
+// StorageClasses are non-namespaced; the name of the storage class
+// according to etcd is in ObjectMeta.Name.
+type StorageClass struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	// ProvisionerType indicates the type of the provisioner.
+	ProvisionerType string `json:"provisionerType,omitempty"`
+
+	// ProvisionerParameters holds the parameters for the provisioner that should
+	// create volumes of this storage class.
+	ProvisionerParameters map[string]string `json:"provisionerParameters,omitempty"`
```

Comment:
The Ceph Volume Plugin needs secrets. We'll need it if we are to add a Ceph
provisioner.

On Tue, Jul 19, 2016 at 10:19 AM, bradley childs <notifications@github.com>
wrote:

> In pkg/apis/extensions/types.go
> <https://github.com/kubernetes/kubernetes/pull/29006#discussion_r71359954>
> :
>
> > +
> > +// StorageClass describes the parameters for a class of storage for
> > +// which PersistentVolumes can be dynamically provisioned.
> > +//
> > +// StorageClasses are non-namespaced; the name of the storage class
> > +// according to etcd is in ObjectMeta.Name.
> > +type StorageClass struct {
> > +	unversioned.TypeMeta `json:",inline"`
> > +	api.ObjectMeta       `json:"metadata,omitempty"`
> > +
> > +	// ProvisionerType indicates the type of the provisioner.
> > +	ProvisionerType string `json:"provisionerType,omitempty"`
> > +
> > +	// ProvisionerParameters holds the parameters for the provisioner that should
> > +	// create volumes of this storage class.
> > +	ProvisionerParameters map[string]string `json:"provisionerParameters,omitempty"`
>
> Since none of our intree provisioners need secrets right now, I think we
> should keep that for the second phase.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29006/files/c28e599c704a45c55eb83426fb761329adcc6ed4#r71359954>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADDZzDn9kq-e3mauq2kRGO8XoLtNsPCaks5qXOsWgaJpZM4JNdmW>
> .
>